[clang] Rename files that MacOS libtool warns about (NFC)

jthackray opened the pull request:
As mentioned in https://discourse.llvm.org/t/rfc-rename-source-files-in-clang-lib-codegen-targetbuiltins/87462/ it appears that MacOS's libtool warns about source filenames that are identically named, even if they exist in separate directories.

Sadly, there doesn't appear to be an easy way to disable this warning, so rename these files, as these warnings are annoying for MacOS users.

Fixes #133199.

farzonl commented:
This change will absolutely fix the issue, I was advised on my few attempts that we want to be able to keep the same name and the fix should be in cmake and not a file rename.

For example We have many `AMDGPU.cpp`
- clang/lib/Basic/Targets/AMDGPU.cpp
- clang/lib/CodeGen/Targets/AMDGPU.cpp
- clang/lib/CodeGen/TargetBuiltins/AMDGPU.cpp

And we want to be able to keep reusing the AMDGPU.cpp name but it requires us to not reuse that name in the same cmake file and to have one cmake file per directory.

farzonl commented on clang/lib/CodeGen/TargetBuiltins/BuiltinAMDGPU.cpp:
```diff
@@ -1,4 +1,4 @@
-//===------- AMDCPU.cpp - Emit LLVM Code for builtins ---------------------===//
+//===------- BuiltinAMDCPU.cpp - Emit LLVM Code for builtins --------------===//
```
```suggestion
//===------- BuiltinAMDGPU.cpp - Emit LLVM Code for builtins --------------===//
```

jthackray committed:
[clang] Rename files that MacOS libtool warns about (NFC)

As mentioned in https://discourse.llvm.org/t/rfc-rename-source-files-in-clang-lib-codegen-targetbuiltins/87462/
it appears that MacOS's libtool warns about source filenames that
are identically named, even if they exist in separate directories.

Sadly, there doesn't appear to be an easy way to disable this warning,
so rename these files, as these warnings are annoying for MacOS users.

Fixes #133199.

jthackray commented:
@farzonl I see you've tried to fix this twice before, and twice reverted CMake changes.

Looking through the man page for `libtool`, it says:
```
       -s     Produce the preferred type of table of contents, which results in faster link editing when linking with the archive.  The order of the table of
              contents is sorted by symbol name.  The library member name of this type of table of contents is ``__.SYMDEF SORTED''.  This type of table of
              contents can only be produced when the library does not have multiple members that define the same symbol.  This is the default.

       -a     Produce the original type of table of contents, whose order is based on the order of the members in the archive.  The library member name of
              this type of table of contents is ``__.SYMDEF''.  This type of table of contents must be used when the library has multiple members that define
              the same symbol.
```
Is it possible that adding `-a` to llvm's `libtool` invocation would remove this warning. Currently, it must be defaulting to `-s` which warns. I don't usually build on a Mac (I build on Graviton4) but I will try building overnight on a Mac to replicate the warning, and see if adding `-a` causes the warning to go away.

yronglin commented:
Should we just use ar/ranlib instead of libtool on Apple platform? I'm not sure how much performance difference there is between ar and libtool.
